app/src/main/java/com/example/internetshop/presentation/activity/MainActivity.kt

```diff
@@ -93,13 +103,22 @@ class MainActivity : AppCompatActivity(), ContainerHolder {
         })
     }
 
-    override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        return when (item.itemId) {
-            R.id.home -> {
-                supportFragmentManager.popBackStack()
-                true
-            }
-            else -> super.onOptionsItemSelected(item)
+    private fun setupAppBar() {
+        WindowCompat.setDecorFitsSystemWindows(window, false)
+        setSupportActionBar(binding?.toolbar)
+        supportActionBar?.setHomeAsUpIndicator(
+            (ResourcesCompat.getDrawable(resources, R.drawable.ic_back_arrow, null))
+        )
+    }
+
+    private fun updateLayoutParams() {
```

Comment:
First of all name like `updateLayoutParams` say nothing what happened inside. Also it confuses a bit, because I decided that you update activity layout params here. Would be better to name it like `applyInsetsToAppBar` or something like this.

Reply:
fixed